router/broker.go

```diff
@@ -33,12 +36,38 @@ type subscription struct {
 	subscribers map[*wamp.Session]struct{}
 }
 
+// storedEvent is a wrapper around wamp event message with timestamp
+// to be used in event history
+type storedEvent struct {
+	timestamp    time.Time
+	Subscription wamp.ID
+	Publication  wamp.ID
+	Details      wamp.Dict
+	Arguments    wamp.List `wamp:"omitempty"`
+	ArgumentsKw  wamp.Dict `wamp:"omitempty"`
+}
+
+type historyEntry struct {
+	publication *wamp.Publish
+	event       storedEvent
+}
+
+type historyStore struct {
+	events         []historyEntry
```

Comment:
If you rename this to `entries` then you won't have the `event.event` stutter down below.

Reply:
👍🏻  Indeed! Thnx!